### ISSUE TITLE
ASP.NET Core SignalR 2.2 Authentication (Windows Authentication) on .net core console app returns - 401 (Unauthorized) 

### ISSUE THREAD

**Issue by zanate4019** [aspnet/AspNetCore#9947]
### Describe the bug
I have a signalr hub (Startup.cs attached).  It is configured for windows authentication.  CORS is configured to be wide open.  It is hosted in IIS.  I have a .Net Core console application that when run on a different server it runs fine.  When it runs on the same server it receives a 401 Unauthorized response.
![image](https://user-images.githubusercontent.com/10452739/57144027-b2d6a480-6d85-11e9-99ba-f7f2c42329f6.png)

### To Reproduce
Relevant Signalr Config for Configure Services
```
// IISDefaults requires the following import:
            // using Microsoft.AspNetCore.Server.IISIntegration;
            services.AddAuthentication(IISDefaults.AuthenticationScheme);

// configure SignalR
            services.AddSignalR( o =>
            {
                o.EnableDetailedErrors = true;

            });

            services.Configure<IISServerOptions>(options =>
            {
                options.AutomaticAuthentication = true; //uses windows authentication
            });
```

Relevant SignalR Config for Configure
``` 
           app.UseSignalR(routes => routes.MapHub<ContactRequestHub>(srConfig.ContactRequestHubUrl, o =>
            {
                o.ApplicationMaxBufferSize = 0;
                o.TransportMaxBufferSize = 0;
            }));
            app.UseSignalR(routes => routes.MapHub<ExamHub>(srConfig.ExamHubUrl, o =>
            {
                o.ApplicationMaxBufferSize = 0;
                o.TransportMaxBufferSize = 0;
            }));
            app.UseSignalR(routes => routes.MapHub<PagingHub>(srConfig.PagingHubUrl, o =>
            {
                o.ApplicationMaxBufferSize = 0;
                o.TransportMaxBufferSize = 0;
            }));
            app.UseSignalR(routes => routes.MapHub<UserHub>(srConfig.UserHubUrl, o =>
            {
                o.ApplicationMaxBufferSize = 0;
                o.TransportMaxBufferSize = 0;
            }));
            app.UseSignalR(routes => routes.MapHub<WorkItemHub>(srConfig.WorkItemHubUrl, o =>
            {
                o.ApplicationMaxBufferSize = 0;
                o.TransportMaxBufferSize = 0;
            }));
```

Relevant Client Config
```        
private void startExamHub()
        {
            try
            {
                _examHubConnection = new HubConnectionBuilder()
                                .WithUrl(_examNotificationUrl,
                                    options => options.UseDefaultCredentials = true)
                                .Build();
                _examHubConnection.Closed += async (error) =>
                {
                    await Task.Delay(new Random().Next(0, 5) * 1000);
                    await _examHubConnection.StartAsync();
                };
                _examHubConnection.StartAsync().Wait();
            }
            catch (Exception ex)
            {
                throw new SystemException("Error starting exam connection", ex);
            }
        }
```

### Expected behavior
Client should be able to connect and send messages both locally and remotely.

### Additional context
.NET Core SDK (reflecting any global.json):
 Version:   2.2.202
 Commit:    8a7ff6789d

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17763
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.2.202\

Host (useful for support):
  Version: 2.2.3
  Commit:  6b8ad509b6

.NET Core SDKs installed:
  2.1.400 [C:\Program Files\dotnet\sdk]
  2.1.401 [C:\Program Files\dotnet\sdk]
  2.1.403 [C:\Program Files\dotnet\sdk]
  2.1.500 [C:\Program Files\dotnet\sdk]
  2.1.502 [C:\Program Files\dotnet\sdk]
  2.1.505 [C:\Program Files\dotnet\sdk]
  2.1.602 [C:\Program Files\dotnet\sdk]
  2.2.100 [C:\Program Files\dotnet\sdk]
  2.2.202 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
[Startup.zip](https://github.com/aspnet/AspNetCore/files/3142345/Startup.zip)

**Comment by BrennanConroy** [comment]
cc @Tratcher

**Comment by Tratcher** [comment]
On the local machine it's going to use NTLM but on the remote machine it's going to default to stronger mechanisms like Kerberos that require a more configuration. Can you share your Windows Auth settings from IIS?
Also, the client settings matter. Are you using default credentials and the full server name?

**Comment by zanate4019** [comment]
Sorry it took so long for me get back -- I've been out of town at a ServiceNow conference.

I am using default creds.  I have tried accessing the Sr hub by netbios name, localhost, and fqdn. Although I'd like it be able to work by CNAME record name.  We typically will hostname filtering on our web servers (its not currently in the mix though).

Is this what you are looking for in IIS:
![image](https://user-images.githubusercontent.com/10452739/57542540-99020800-7317-11e9-89f5-8e39b4b45ff9.png)

**Comment by zanate4019** [comment]
I take that back...It works using FQDN.  Strange, in my past experience (which is admittedly dated), NTLM was always more allowing than Kerberos.  Any idea why it would work remotely using an alias (CNAME), but not locally.  It also doesnt work locally using localhost.

**Comment by anurse** [comment]
Since it works using FQDN this seems beyond our scope since we don't have that level of control over the process. We're just getting Windows/IIS to do the authentication. Perhaps Windows Support is a better avenue to explore here.

**Comment by zanate4019** [comment]
Might be, but it definitely works with the previous version of SignalR.  I know the architecture is completely different, but IIS and Windows are the same.

**Comment by zanate4019** [comment]
I found a workaround.  I'm a little leary about this being the solution for a production system, but I thought I would post it for comment to see.

I was looking for a solution from a Windows standpoint and found the following article:
https://social.technet.microsoft.com/Forums/windowsserver/en-US/ee6d2e78-6bab-42c2-a709-04df66e7979f/navigating-to-a-server-alias-from-itself-prompts-for-credentials-windows-server-2008-r2?forum=winserverNIS

This post on TechNet didn't have anything to do with SignalR or web applications but did mention a server that was not able to navigate to itself via CNAME.

To summarize the post it said the following was recommended:

- Allowing other machines to use filesharing via the DNS Alias (DisableStrictNameChecking)
- Allowing server machine to use filesharing with itself via the DNS Alias (BackConnectionHostNames) 
- Providing browse capabilities for multiple NetBIOS names (OptionalNames)
- Register the Kerberos service principal names (SPNs) for other Windows functions like Printing (setspn) 

I tried all 4 and it worked.  I'm not sure if all 4 are necessary as I haven't yet tried different combinations.

**Comment by DAllanCarr** [comment]
@zanate4019 

> I tried all 4 and it worked. I'm not sure if all 4 are necessary as I haven't yet tried different combinations.

This would be a good candidate for docs. 😉

**Comment by zanate4019** [comment]
I applied the same work around to the Windows 2008 R2 server and it worked there as well.

**Comment by anurse** [comment]
Are you using in-proc or out-of-proc hosting for IIS? Windows Authentication for the Out-Of-Proc model is a bit tricky since it involves passing handles across the process boundary. If you are using Out-Of-Proc, does the same issue repro if you switch to In-Proc? (see docs on in-proc/out-of-proc here: https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/iis/?view=aspnetcore-2.2#application-configuration)

**Comment by zanate4019** [comment]
It's in-proc.

**Comment by anurse** [comment]
The actual authentication behavior in IIS between ASP.NET 4.x and ASP.NET Core is identical and lives in IIS (not ASP.NET Core). I'm not sure there's much we can do here. Is there a docs change that you could suggest that would have helped you identify the solution you found above?

**Comment by anurse** [comment]
Closing this as we haven't heard from you and generally close issues with no response after ~7 days.
